pkg/controllers/clusterresourceplacement/placement_status_test.go

```diff
@@ -909,49 +968,1282 @@ func TestSetPlacementStatus(t *testing.T) {
+						},
+						{
+							ClusterName: cluster2Name,
+							Selected:    true,
+							Reason:      "success",
+						},
+					},
+				},
+			},
+			latestResourceSnapshot: &fleetv1beta1.ClusterResourceSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fmt.Sprintf(fleetv1beta1.ResourceSnapshotNameFmt, testName, 0),
+					Labels: map[string]string{
+						fleetv1beta1.ResourceIndexLabel:    "0",
+						fleetv1beta1.CRPTrackingLabel:      testName,
+						fleetv1beta1.IsLatestSnapshotLabel: "true",
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							Name:               testName,
+							BlockOwnerDeletion: pointer.Bool(true),
+							Controller:         pointer.Bool(true),
+							APIVersion:         fleetAPIVersion,
+							Kind:               "ClusterResourcePlacement",
+						},
+					},
+					Annotations: map[string]string{
+						fleetv1beta1.ResourceGroupHashAnnotation:         "hash",
+						fleetv1beta1.NumberOfResourceSnapshotsAnnotation: "1",
+					},
+				},
+			},
+			clusterResourceBindings: []fleetv1beta1.ClusterResourceBinding{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "binding-1",
+						Labels: map[string]string{
+							fleetv1beta1.CRPTrackingLabel: testName,
+						},
+						Generation: 1,
+					},
+					Spec: fleetv1beta1.ResourceBindingSpec{
+						State:                        fleetv1beta1.BindingStateBound,
+						ResourceSnapshotName:         fmt.Sprintf(fleetv1beta1.ResourceSnapshotNameFmt, testName, 0),
+						SchedulingPolicySnapshotName: fmt.Sprintf(fleetv1beta1.PolicySnapshotNameFmt, testName, 0),
+						TargetCluster:                cluster1Name,
+					},
+					Status: fleetv1beta1.ResourceBindingStatus{
+						Conditions: []metav1.Condition{
+							{
+								Status:             metav1.ConditionFalse,
+								Type:               string(fleetv1beta1.ResourceBindingBound),
+								Reason:             "resourceBindingTrue",
+								ObservedGeneration: 1,
+								LastTransitionTime: oldTransitionTime,
+							},
+						},
+					},
+				},
+			},
+			wantStatus: &fleetv1beta1.ClusterResourcePlacementStatus{
+				SelectedResources: selectedResources,
+				Conditions: []metav1.Condition{
+					{
+						Status:             metav1.ConditionUnknown,
+						Type:               string(fleetv1beta1.ClusterResourcePlacementAppliedConditionType),
+						Reason:             ApplyPendingReason,
+						ObservedGeneration: crpGeneration,
+						LastTransitionTime: oldTransitionTime,
+					},
+					{
+						Status:             metav1.ConditionTrue,
+						Type:               string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType),
+						Reason:             "Scheduled",
+						ObservedGeneration: crpGeneration,
+						LastTransitionTime: oldTransitionTime,
+					},
+					{
+						Status:             metav1.ConditionFalse,
+						Type:               string(fleetv1beta1.ClusterResourcePlacementSynchronizedConditionType),
+						Reason:             SynchronizePendingReason,
```

Comment:
why is the reason still "pending" when one of the resource binding status is False already?

Reply:
i think we discussed this before. The conclusion is, if the sync has not finished, we always set the appliedCondition as unknown. Until the sync is finished, we will set the appliedCondition as true or false.

It makes the documentation easy otherwise there will be a lot of combinations and too complicated for the user.